lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -326,14 +326,16 @@ class ModuleEmitter : public VerilogEmitterBase,
   LogicalResult visitSV(AssignInterfaceSignalOp op);
   void emitOperation(Operation *op);
 
+  using ValueOrOp = PointerUnion<Value, Operation *>;
+
   void collectNamesEmitDecls(Block &block);
-  StringRef addName(Value value, StringRef name);
-  StringRef addName(Value value, StringAttr nameAttr) {
-    return addName(value, nameAttr ? nameAttr.getValue() : "");
+  StringRef addName(ValueOrOp valueOrOp, StringRef name);
+  StringRef addName(ValueOrOp valueOrOp, StringAttr nameAttr) {
+    return addName(valueOrOp, nameAttr ? nameAttr.getValue() : "");
   }
 
-  StringRef getName(Value value) {
-    auto *entry = nameTable[value];
+  StringRef getName(ValueOrOp valueOrOp) {
```

Comment:
This uglifies a lot of the callers which is unfortunate.  I'd recommend adding an overload to help with this:

`StringRef getName(Value value) { return getName(ValueOrOp(value)); }`

This will remove a lot of the changes later in the file.  The instance stuff can use the general one, we don't need an overload for Operation*.

Reply:
I added this overload, but in fact I only found one caller that was improved by this. @lattner since you were talking about removing a lot of changes, I'm not sure if we are on the same page. Was there something else to clean up here?

A lot of the changes in this file are `value`->`valueOrOp`, because I renamed the formal parameter to `addName` to be more descriptive/less misleading. Are these the changes you were referring to?

Reply:
aha, ok if it only helps in one place, then please feel free to drop it!  Sorry for the distraction